### PR TITLE
Add types to shortcut

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Description
Add types to `Shortcut` and update type-refinements in the component.

## How has this been tested?
Local typecheck.

## Types of changes
Type updates.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. 

/cc @sirreal

### CODE REVIEW

File: packages/components/src/shortcut/index.js [hunk 1]
```diff
@@ -13,9 +24,7 @@ function Shortcut( { shortcut, className } ) {
 
 	if ( isString( shortcut ) ) {
 		displayText = shortcut;
-	}
-
-	if ( isObject( shortcut ) ) {
```
Comment (sirreal) [inline packages/components/src/shortcut/index.js:18]: This change concerns me a bit… we only expect to receive a string or object, but this protected against other values before and doesn't after this change.

How do you feel about reverting it?

Comment (saramarcondes) [reply packages/components/src/shortcut/index.js:18]: I will becuase I cannot reproduce the reason I made this change to begin with 🤦‍♀️ There was some issue with refining the type down to an object with the two properties, but it's working now, go figure.

File: packages/components/src/shortcut/index.js [hunk 2]
```diff
@@ -1,8 +1,19 @@
...
+/** @typedef {string | { display: string, ariaLabel: string }} Shortcut */
+/**
+ * @typedef Props
+ * @property {Shortcut} shortcut Shortcut configuration
+ * @property {string} [className] Classname
+ */
+
+/**
+ * @param {Props} props Props
+ * @return {import('react').ReactNode} Element
```
[the remark this answers is not in the document]
Comment (saramarcondes) [reply packages/components/src/shortcut/index.js:15]: I thought `ReactNode` was preferable as a single type, I didn't realize what the exact interface functional components adhered to. I'll change this back to `JSX.Element | null` 👍

**Comment by sirreal** [comment]
This looks good but it needs a rebase.

**Comment by saramarcondes** [comment]
Closing as duplicate of https://github.com/WordPress/gutenberg/pull/26078

**Comment by sarayourfriend** [comment]
I'll reopen this PR in a second.

**Pull request closed without merging**